### DISCUSSION TITLE
how to use caffeine cache when we import spring-boot-starter-data-redis

### DISCUSSION THREAD

**Discussion by hxxgy** [MarcGiffing/bucket4j-spring-boot-starter#106] [Q&A] [answered]
the version of bucket4j-spring-boot-starter is 0.6.1.
when we import spring-boot-starter-data-redis, then if we set bucket4j cache to caffeine, there will be some issue.

```
Parameter 3 of constructor in com.giffing.bucket4j.spring.boot.starter.config.servlet.Bucket4JAutoConfigurationServletFilter required a single bean, but 2 were found:
	- bucket4jCacheResolver: defined by method 'bucket4jCacheResolver' in class path resource [com/giffing/bucket4j/spring/boot/starter/config/cache/jcache/JCacheBucket4jConfiguration.class]
	- bucket4RedisResolver: defined by method 'bucket4RedisResolver' in class path resource [com/giffing/bucket4j/spring/boot/starter/config/cache/redis/RedisBucket4jConfiguration.class]
```

is there any method to solve this problem?
I try to exclude this component by excludeFilters, but it doesn't work.

**Comment by MarcGiffing** [comment, marked as answer]
With the Version 0.0.7 you can set the Cache which should be used with the property

> bucket4j.cache-to-use=jcache

Alternativly you can set the cache to use to something invalid and provide the sync or async Cache Manager by your own:

> bucket4j.cache-to-use=invalid

https://github.com/MarcGiffing/bucket4j-spring-boot-starter/issues/90
